src/config.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  projectId?: string;
```

Comment:
Cloud Trace's config [does not carry the project ID in it](https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/20e69f23d3da2f405f40cbe16df606159d199fa0/config.ts#L21), accepting that via an environment variable instead. Should we do the same as Cloud Trace or just include the project ID in the config, deriving it from the environment when possible (such as in GCE or GAE environment)? I'd prefer the latter.

Reply:
This was WIP, sorry. I needed a single ts file to kick start the migration process. I threw this together.

I think we should follow the same type definition as trace and debug. The actual configuration type is defined by the `@google-cloud/common` module (which is in JS at the moment). Trace, Debug, and this module, each independently extend the configuration interface. The optional `projectId` property should exist in the base class defined by `common` (once it migrates to TS). Each of Trace, Debug and Profiler should extend the interface with their own extensions for config. /cc @lukesneeringer.